wcfsetup/install/files/lib/system/worker/RegisteredWorker.class.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace wcf\system\worker;
+
+use wcf\data\object\type\ObjectType;
+use wcf\system\WCF;
+
+/**
+ * Represents a worker that is registered with the RebuildWorkerCollecting event.
+ *
+ * @author Tim Duesterhus
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Worker
+ * @since 6.0
+ */
+final class RegisteredWorker
+{
+    public function __construct(
+        public readonly string $classname,
+        private readonly ?ObjectType $legacyObjectType = null,
+    ) {
+    }
+
+    public function getName(): string
```

Comment:
I would like to see a `private getIdentifier(): string` that returns the `objectType` or the sanitized `$className`. This allows both `getName()` and `getDescription()` to stop caring about the type of worker and simplify their code to a plain `return`.